Lib/test/test_free_threading/test_type.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+
+from threading import Thread
+from multiprocessing.dummy import Pool
+from unittest import TestCase
+
+from test.support import is_wasi
+
+
+NTHREADS = 6
+BOTTOM = 0
+TOP = 1000
+ITERS = 100
+
+class A:
+    attr = 1
+
+@unittest.skipIf(is_wasi, "WASI has no threads.")
```

Comment:
Oh, probably better to use `@threading_helper.requires_working_threading()`

Reply:
iOS and Android do both have working threading, but they don't support subprocesses. In theory `multiprocessing.dummy.Pool` should work on these platforms, but in practice it doesn't because it imports too much of the main multiprocessing module.

The simplest solution is probably to use `concurrent.futures.ThreadPoolExecutor` instead.